src/Plugin/Commerce/PaymentGateway/OnsiteInterface.php

```diff
@@ -16,4 +16,16 @@
 interface OnsiteInterface extends
   OnsitePaymentGatewayInterface,
   SupportsRefundsInterface,
-  SupportsNotificationsInterface {}
+  SupportsNotificationsInterface {
+
+  /**
+   * Indicates the remote payment method name for Card transactions.
+   */
+  const REMOTE_PAYMENT_METHOD_NAME_CC = 'CC';
+
+  /**
+   * Indicates the remote payment method name for ECP transactions.
+   */
+  const REMOTE_PAYMENT_METHOD_NAME_ECP = 'ECP';
```

Comment:
@shabananavas From docs (https://support.bluesnap.com/docs/ipn-parameter-reference):

> Method of payment (i.e. CC/Paypal/Wire).

Have you tested ECP i.e. that we are getting ECP as the payment method's name and not Wire as per docs?

Reply:
Yeah, I saw that doc but I found another doc (which I can't find now) which had ECP as one of the paymentMethods. We can't really test the check payments on the site because when the order is created the payment is registered as 'Pending'. And there is nothing on their account to simulate an IPN that puts the particular check payment in completed mode so that we can investigate the IPN.

However, I did generate a 'Decline' IPN to our test site:

![image](https://user-images.githubusercontent.com/11439056/67861447-4119c080-fb46-11e9-8505-e42f2c5398d9.png)

And this is what's in the IPN:

![image](https://user-images.githubusercontent.com/11439056/67861562-77efd680-fb46-11e9-890c-394528baa6d9.png)

So we do get 'ECP' as the paymentMethod name. So, I'm pretty sure it's 'ECP'.